tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testName           = "volumeproperty"
+	diskStatusDetached = "detached"
+	size               = "100MB"
+	diskFormatOption   = " -o diskformat="
+)
+
+type VolumePropertyTestSuite struct {
+	volumeNames           []string
+	containerName         string
+	formatTypes           []string
+	dockerVolumeRemoveCmd string
+	volumeAttached        bool
+	config                *inputparams.TestConfig
+}
+
+func (s *VolumePropertyTestSuite) SetUpTest(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping volumeproperty_test.TestVolumeProperties.")
+	}
+	s.containerName = inputparams.GetContainerNameWithTimeStamp(testName)
+	s.formatTypes = []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	s.dockerVolumeRemoveCmd = dockerconst.RemoveVolume
+	log.Println(" SETUP Finished: volumeproperty_test.TestVolumeProperties")
+}
+
+// TearDownTest - Removes all the containers and deletes the volumes
+func (s *VolumePropertyTestSuite) TearDownTest(c *C) {
+	out, err := dockercli.RemoveAllContainers(s.config.DockerHosts[1])
+	c.Assert(err, IsNil, Commentf(out))
+	s.dockerVolumeRemoveCmd = s.dockerVolumeRemoveCmd + strings.Join(s.volumeNames, " ")
+	out, err = ssh.InvokeCommand(s.config.DockerHosts[1], s.dockerVolumeRemoveCmd)
+	c.Assert(err, IsNil, Commentf(out))
+	log.Println(" CLEAN-UP Finished: volumeproperty_test.TestVolumeProperties")
+}
+
+var _ = Suite(&VolumePropertyTestSuite{})
+var dockerCliMap = make(map[string][]string)
+var adminCliMap = make(map[string][]string)
+var expectedValues = make(map[string][]string)
+
+/*
+Test steps:
+1. SSH to a vm and create volumes of size 100 mb and specified disk format ["thin", "zeroedthick", "eagerzeroedthick"]
+2. Do docker inspect on the volume get size, disk format and attached-to-vm field.
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", no-value} respectively.
+3. SSH to the esx and get capacity, disk format and attached-to-vm field for the volume using Admin cli
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", "detached"} respectively.
+4. Verifies values obtained from admin cli and docker cli.
+5. SSH to the vm and run a container and mount the volume created.
+6. SSH to the vm and esx and verify the attached-to-vm field for volume - both docker cli and admin cli values should be same.
+7. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+
+NOTE: Do steps 5 ,6 and 7 only for volume of 'thin' disk format
+*/
+
+func (s *VolumePropertyTestSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: volumeproperty_test.TestVolumeProperties")
```

Comment:
should be replaced with https://github.com/vmware/docker-volume-vsphere/blob/master/tests/utils/misc/misc.go#L53